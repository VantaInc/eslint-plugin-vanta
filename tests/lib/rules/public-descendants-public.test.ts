import {
  GraphQLRuleTester,
  ParserOptions,
} from "@graphql-eslint/eslint-plugin";
import rule from "../../../lib/rules/public-descendants-public";

const ruleTester = new GraphQLRuleTester();

const useSchema = (
  schema: string
): { code: string; parserOptions: ParserOptions } => {
  return {
    parserOptions: { schema },
    code: schema,
  };
};

ruleTester.runGraphQLTests("public-descendants-public", rule, {
  valid: [
    useSchema(`
type T1 {
  val: T2 @public
}

type T2 @public {
  val: String
}
`),
    useSchema(`
type T1 {
  val: T2
}

type T2 @public {
  val: String
}
`),
    useSchema(`
type T1 {
  val: T2
}

type T2 {
  val: String
}
`),
    useSchema(`
type T1 {
  val: T2 @public
}

type T2 @public {
  val: T3
}

type T3 @public {
  val: String
}
`),

    useSchema(`
type T1 {
  val: T2
}

type T2 {
  val: T3 @public
}

type T3 @public {
  val: String
}
`),
    useSchema(`
type T1 {
  val: T2
}

type T2 {
  val: T3 @public
}

type T3 @public {
  val: String
}
`),
    useSchema(`
type T1 @public {
  val: T2
  val2: T3
}

type T2 @public {
  val: String
}

type T3 @public {
  val: String
}
`),
  ],
  invalid: [
    {
      ...useSchema(`
type T1 {
  val: T2 @public
}

type T2 {
  val: String
}
`),
      errors: 1,
    },
    {
      ...useSchema(`
type T1 {
  val: T2 @public
}

type T2 {
  val: String @public
}
`),
      errors: 1,
    },
    {
      ...useSchema(`
type T1 {
  val: T2 @public
}

type T2 @public {
  val: T3
}

type T3 {
  val: String
}
`),
      errors: 1,
    },
    {
      ...useSchema(`
type T1 {
  val: T2
}

type T2 {
  val: T3 @public
}

type T3 {
  val: String
}
`),
      errors: 1,
    },
  ],
});
