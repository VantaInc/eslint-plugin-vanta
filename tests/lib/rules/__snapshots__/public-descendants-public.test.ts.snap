// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Invalid #1 1`] = `
"#### ⌨️ Code

       1 |   directive @public(
       2 |     detail: String = "No authentication required for this action"
       3 |   ) on FIELD_DEFINITION | OBJECT | UNION | ENUM
       4 |
       5 |   type T1 {
       6 |     val: T2 @public
       7 |   }
       8 |
       9 |   type T2 {
      10 |     val: String
      11 |   }

#### ❌ Error

      5 |   type T1 {
    > 6 |     val: T2 @public
        |     ^^^^^^^^^ Types accessible by @public fields must be marked @public.
      7 |   }"
`;

exports[`Invalid #2 1`] = `
"#### ⌨️ Code

       1 |   directive @public(
       2 |     detail: String = "No authentication required for this action"
       3 |   ) on FIELD_DEFINITION | OBJECT | UNION | ENUM
       4 |
       5 |   type T1 {
       6 |     val: T2 @public
       7 |   }
       8 |
       9 |   type T2 {
      10 |     val: String @public
      11 |   }

#### ❌ Error

      5 |   type T1 {
    > 6 |     val: T2 @public
        |     ^^^^^^^^^ Types accessible by @public fields must be marked @public.
      7 |   }"
`;

exports[`Invalid #3 1`] = `
"#### ⌨️ Code

       1 |   directive @public(
       2 |     detail: String = "No authentication required for this action"
       3 |   ) on FIELD_DEFINITION | OBJECT | UNION | ENUM
       4 |
       5 |   type T1 {
       6 |     val: T2 @public
       7 |   }
       8 |
       9 |   type T2 @public {
      10 |     val: T3
      11 |   }
      12 |
      13 |   type T3 {
      14 |     val: String
      15 |   }

#### ❌ Error

       8 |
    >  9 |   type T2 @public {
         |   ^^^^^^^^^^^^^^^^^
    > 10 |     val: T3
         | ^^^^^^^^^^^
    > 11 |   }
         | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Types accessible from @public types must themselves be marked @public. Unmarked field: val
      12 |"
`;

exports[`Invalid #4 1`] = `
"#### ⌨️ Code

       1 |   directive @public(
       2 |     detail: String = "No authentication required for this action"
       3 |   ) on FIELD_DEFINITION | OBJECT | UNION | ENUM
       4 |
       5 |   type T1 {
       6 |     val: T2
       7 |   }
       8 |
       9 |   type T2 {
      10 |     val: T3 @public
      11 |   }
      12 |
      13 |   type T3 {
      14 |     val: String
      15 |   }

#### ❌ Error

       9 |   type T2 {
    > 10 |     val: T3 @public
         |     ^^^^^^^^^ Types accessible by @public fields must be marked @public.
      11 |   }"
`;
